CMakeLists.txt

```diff
@@ -13,10 +13,42 @@ TRIBITS_PACKAGE_DECL(DataTransferKit)
 TRIBITS_ADD_DEBUG_OPTION()
 TRIBITS_ADD_SHOW_DEPRECATED_WARNINGS_OPTION()
 
-TRIBITS_ADD_ETI_INSTANTIATIONS(TpetraCore
-  "S={double} LO={int} GO={unsigned long} N={Kokkos::Compat::KokkosSerialWrapperNode}"
-  )
+##---------------------------------------------------------------------------##
+## Explicit template instantiation
+##---------------------------------------------------------------------------##
+
+# Setup the same ETI as Tpetra but do not set global ordinal types as
+# aditional scalar types.
+
+# check tpetra variabls are defined.
+ASSERT_DEFINED(TpetraCore_ETI_SCALARS)
+ASSERT_DEFINED(TpetraCore_ETI_LORDS)
+ASSERT_DEFINED(TpetraCore_ETI_GORDS)
+ASSERT_DEFINED(TpetraCore_ETI_NODES)
+
+# set the eti types
+SET (DataTransferKit_ETI_SCALARS_TEMP ) # start with the empty list
```

Comment:
I understood what the loop was doing.  My point was you don't really need the `*_TEMP` variable, you could be directly filling `DataTransferKit_ETI_SCALARS`, but that's alright.